src/model/Document.cpp

```diff
@@ -407,13 +407,6 @@ auto Document::operator=(const Document& doc) -> Document& {
     buildContentsModel();
     updateIndexPageNumbers();
 
-    bool lastLock = tryLock();
-    unlock();
-    this->handler->fireDocumentChanged(DOCUMENT_CHANGE_COMPLETE);
-    if (!lastLock)  // document was locked before
```

Comment:
Now we lock again if we had the lock before calling fireDocumentChanged on the assumption that the behavior of tryLock is defined. Since this is not the case this whole code seems really weird. I don't know if removing all of this breaks anything else but in my testing everything has been fine.